packages/lab/src/date-input/DateInput.tsx

```diff
@@ -153,30 +181,78 @@ export const DateInput = forwardRef<HTMLDivElement, DateInputProps>(
       ...restDateInputProps
     } = inputProps;
 
+    // Effects. Update date strings when dates change
+    useEffect(() => {
```

Comment:
would useCallback work here instead ?

Reply:
because `startDateStingValue` needs to change on every stroke in the input `onChange`, it can't just follow the calendar value